blocks/extensions/saveload.py

```diff
@@ -85,83 +84,78 @@ def do(self, callback_name, *args):
         construction stage.
 
         """
-        from_main_loop, from_user = self.parse_args(callback_name, args)
+        _, from_user = self.parse_args(callback_name, args)
         try:
             path = self.path
             if from_user:
                 path, = from_user
-            already_saved_to = self.main_loop.log.current_row.get(SAVED_TO, ())
-            self.main_loop.log.current_row[SAVED_TO] = (
-                already_saved_to + (path,))
-            secure_pickle_dump(self.main_loop, path)
+            secure_dump(self.main_loop, path)
```

Comment:
Right, I actually started on that but wanted to keep it for later, I guess I might as well finish it. I'll commit it in a few minutes.